### PR TITLE
aspen cli db - create mega fasta

### PR DESCRIPTION
**Pull request by phoenixAja**
### Description

add aspen cli command to create mega fasta from input file of public identifiers

#### Issue
no issue

### Test plan

wrote a test

### CODE REVIEW

File: src/backend/aspen/cli/db.py [hunk 1]
```diff
@@ -408,3 +410,54 @@ def create_phylo_run(
...
+    engine = ctx.obj["ENGINE"]
+    session = engine.make_session()
+
+    public_identifiers: Collection[str] = [
+        line.strip() for line in public_identifier_input_fh
+    ]
+
+    all_samples: Iterable[Sample] = (
+        session.query(Sample)
+        .filter(Sample.public_identifier.in_(public_identifiers))
```
Comment (ttung) [inline src/backend/aspen/cli/db.py]:
```suggestion
        .filter(
            and_(
                Sample.uploaded_pathogen_genome != None,  # noqa: E711
                Sample.public_identifier.in_(public_identifiers))
            )
        )
```

File: src/backend/aspen/cli/db.py [hunk 2]
```diff
@@ -408,3 +410,54 @@ def create_phylo_run(
...
+@click.option(
+    "sequences_output_fh", "--sequences-output", type=click.File("w"), required=True
+)
+@click.pass_context
+def create_mega_fasta(
+    ctx, public_identifier_input_fh: io.TextIOBase, sequences_output_fh: io.TextIOBase
+):
+
+    engine = ctx.obj["ENGINE"]
+    session = engine.make_session()
```
Comment (ttung) [inline src/backend/aspen/cli/db.py]:
```suggestion
    with session_maker(engine) as session:
```

File: src/backend/aspen/cli/db.py [hunk 3]
```diff
@@ -408,3 +410,54 @@ def create_phylo_run(
...
+
+    public_identifiers: Collection[str] = [
```
Comment (ttung) [inline src/backend/aspen/cli/db.py]:
```suggestion
    public_identifiers: Sequence[str] = [
```

sqlalchemy's `in_` works differently than python's `in` sadly.

File: src/backend/aspen/cli/db.py [hunk 4]
```diff
@@ -408,3 +410,54 @@ def create_phylo_run(
...
+        .options(
+            joinedload(Sample.uploaded_pathogen_genome).undefer(
+                UploadedPathogenGenome.sequence
+            ),
+        )
+    )
+
+    # TODO: this code is now used in multiple places, find central location to store this
+    for sample in all_samples:
+        # filter for samples that passed genome recovery
```
Comment (ttung) [inline src/backend/aspen/cli/db.py]: consider a TODO: for supporting called pathogen genomes (or an explanation of why it's not necessary)

File: src/backend/aspen/cli/tests/test_db_cli.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+from aspen.cli import db
+from aspen.database.connection import SqlAlchemyInterface
+from aspen.workflows.pangolin.tests.test_pangolin_workflow import create_test_data
+
+
+def test_create_mega_fasta(mocker, session, postgres_database):
+
+    create_test_data(session)
+    public_identifiers_txt: PosixPath = Path(
```
Comment (ttung) [inline src/backend/aspen/cli/tests/test_db_cli.py]:
```suggestion
    public_identifiers_txt = Path(
```

someone could run this on Windows. 😮

**Review by ttung — APPROVED** [review]
see comments, rest lgtm.

File: src/backend/aspen/cli/tests/test_db_cli.py (hunk 5 above)
Comment (phoenixAja) [reply src/backend/aspen/cli/tests/test_db_cli.py]: lololol

File: src/backend/aspen/cli/db.py (hunk 4 above)
Comment (phoenixAja) [reply src/backend/aspen/cli/db.py]: ok i will add a todo for now

File: src/backend/aspen/cli/db.py (hunk 3 above)
Comment (phoenixAja) [reply src/backend/aspen/cli/db.py]: ah interesting, Why Sequence and not Iterable or list type hint here?

Comment (ttung) [reply src/backend/aspen/cli/db.py]: Generally speaking, variables and parameters should be declared with the lowest type possible (to provide flexibility).  There's a school of thought that return values should be declared with the highest type possible.  I'm not sure I subscribe to that, so happy to debate that one. :)

**Comment by phoenixAja** [comment]
crap i should have added you as a co-author @ttung, totally was spacing..

**Pull request merged**